src/gui/tracks/PatternTrackView.cpp

```diff
@@ -78,5 +84,65 @@ void PatternTrackView::clickedTrackLabel()
 	getGUI()->patternEditor()->setFocus(Qt::ActiveWindowFocusReason);
 }
 
+QMenu* PatternTrackView::createMixerMenu(QString title, QString newMixerLabel)
+{
+	// We have a different title for Pattern Tracks
+	QMenu* mixerMenu = new QMenu(tr("Assign all to mixer channel"));
+
+	mixerMenu->addAction(newMixerLabel, this, SLOT(createMixerLine()));
+	mixerMenu->addSeparator();
+
+	for (int i = 0; i < Engine::mixer()->numChannels(); ++i)
+	{
+		MixerChannel* curChannel = Engine::mixer()->mixerChannel(i);
+
+		auto index = curChannel->m_channelIndex;
+
+		QString label = tr("%1: %2").arg(index).arg(curChannel->m_name);
+
+		mixerMenu->addAction(label, [this, index](){
+			assignMixerLine(index);
+		});
+	}
+
+	return mixerMenu;
+}
+
+void PatternTrackView::createMixerLine()
+{
+	int channelIndex = getGUI()->mixerView()->addNewChannel();
+	auto channel = Engine::mixer()->mixerChannel(channelIndex);
+
+	channel->m_name = getPatternTrack()->name();
+	if (getTrack()->useColor()) { channel->setColor(getTrack()->color()); }
+
+	assignMixerLine(channelIndex);
+}
+
+void PatternTrackView::assignMixerLine(int channelIndex)
+{
+	// Assign all tracks to channel:
+	TrackContainer::TrackList tl = Engine::patternStore()->tracks();
+
+	for
+	(
+		TrackContainer::TrackList::iterator it = tl.begin();
+		it != tl.end();
+		++it
+	)
```

Comment:
```suggestion
	for (Track* track : tl)
```
Instances of `(*it)->` would need to be changed to `track->`, and `*it` changed to `track`.